### DISCUSSION TITLE
oracledb issue after follwoing 2 methods { "errorMessage":  'lambda_function': cannot import name 'base_impl'

### DISCUSSION THREAD

**Discussion by oshaye3** [oracle/python-oracledb#347] [General]
I have a task to connect to an AWS RDS Oracle server and retrieve information using lambda layer. I have been working on this issue for almost two days. I discovered a lightweight Oracle database installation called python-oracledb but haven't been able to resolve the problem. I need help.

I am using Windows on my local machine, and the Oracle RDS is on a Windows EC2 instance as well.

Here is the error I am encountering:

`"Unable to import module 'lambda_function': cannot import name 'base_impl' from partially initialized module 'oracledb' (most likely due to a circular import) (/opt/python/lib/python3.12/site-packages/oracledb/__init__.py)", "errorType": "Runtime.ImportModuleError", "requestId": "0beb35cc-16a9-451a-b13c-b21e5d58a12d", "stackTrace": [] } I`

**Comment by anthony-tuininga** [comment]
If you are able to [build from source](https://python-oracledb.readthedocs.io/en/latest/user_guide/installation.html#installing-from-source-code) you can try the latest code which should supply a better error message. Frequently this error on Windows is due to a missing redistributable, missing permissions or some other misconfiguration -- but the method that was used before to perform the various imports covered that up. Hopefully the new code lets you see the real error message!
